File: app/src/main/java/in/kashewdevelopers/mirrortext/MainActivity.java
package in.kashewdevelopers.mirrortext;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    EditText input;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        input = findViewById(R.id.input);

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        super.onCreateOptionsMenu(menu);

        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu, menu);

        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {

        switch (item.getItemId()) {
            case R.id.about:
                startActivity(new Intent(this, About.class));
                return true;
            case R.id.horizontal:
                item.setChecked(!item.isChecked());
                input.setRotationX(item.isChecked() ? 180 : 0);
                return true;
            case R.id.vertical:
                item.setChecked(!item.isChecked());
                input.setRotationY(item.isChecked() ? 180 : 0);
                return true;
            default:
                return false;
        }

    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        moveTaskToBack(true);
    }
}

File: app/src/main/java/in/kashewdevelopers/mirrortext/MainActivity.kt
package `in`.kashewdevelopers.mirrortext

import `in`.kashewdevelopers.mirrortext.databinding.ActivityMainBinding
import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        super.onCreateOptionsMenu(menu)
        menuInflater.inflate(R.menu.menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.about -> {
                startActivity(Intent(this, About::class.java))
                return true
            }

            R.id.horizontal -> {
                item.isChecked = !item.isChecked
                binding.input.rotationX = if (item.isChecked) 180f else 0f
                return true
            }

            R.id.vertical -> {
                item.isChecked = !item.isChecked
                binding.input.rotationY = if (item.isChecked) 180f else 0f
                return true
            }

            else -> return false
        }
    }

}
